R/plot_taxa.R

```diff
@@ -265,69 +274,76 @@ plot_taxa_accum_time <- function(
 #' \dontrun{
 #' # plot richness through time for ecocomDP formatted dataset by 
 #' # observation date
-#' plot_taxa_richness(
-#'   dataset = ants_L1)
+#' plot_taxa_diversity(ants_L1)
 #' 
 #' # plot richness through time for ecocomDP formatted dataset by 
 #' # aggregating observations within a year
-#' plot_taxa_richness(
-#'   dataset = ants_L1,
+#' plot_taxa_diversity(
+#'   data = ants_L1,
 #'   time_window_size = "year")
 #' 
 #' # plot richness through time for ecocomDP observation table
-#' plot_taxa_richness(
-#'   observation = ants_L1[[1]]$tables$observation)
+#' plot_taxa_diversity(ants_L1[[1]]$tables$observation)
 #' 
 #' # plot richness through time for flattened ecocomDP dataset 
-#' plot_taxa_richness(
-#'   flat_data = flatten_dataset(ants_L1))
+#' plot_taxa_diversity(flatten_data(ants_L1))
 #' 
 #' 
 #' # Using Tidy syntax:
 #' # plot ecocomDP formatted dataset richness through time by 
 #' # observation date
-#' ants_L1 %>% plot_taxa_richness()
+#' ants_L1 %>% plot_taxa_diversity()
 #' 
-#' ants_L1 %>% plot_taxa_richness(time_window_size = "day")
+#' ants_L1 %>% plot_taxa_diversity(time_window_size = "day")
 #' 
 #' # plot ecocomDP formatted dataset richness through time 
 #' # aggregating observations within a month
-#' ants_L1 %>% plot_taxa_richness(time_window_size = "month")
+#' ants_L1 %>% plot_taxa_diversity(time_window_size = "month")
 #' 
 #' # plot ecocomDP formatted dataset richness through time 
 #' # aggregating observations within a year
-#' ants_L1 %>% plot_taxa_richness(time_window_size = "year")
+#' ants_L1 %>% plot_taxa_diversity(time_window_size = "year")
 #' 
 #' # tidy syntax, filter data by date
 #' ants_L1 %>% 
-#'   flatten_dataset() %>% 
+#'   flatten_data() %>% 
 #'   dplyr::filter(
 #'     lubridate::as_date(datetime) > "2007-01-01") %>%
-#'   plot_taxa_richness(
-#'     flat_data = .,
+#'   plot_taxa_diversity(
 #'     time_window_size = "year")
 #' }
-plot_taxa_richness <- function(
-  dataset = NULL,
-  flat_data = NULL,
-  observation = NULL, 
+#' 
+plot_taxa_diversity <- function(
```

Comment:
@sokole Add deprecation notice and handling for observation param. Note this requires observation to be in the set of listed params (i.e. `plot_taxa_richness <- function(data, id = NA_character_, time_window_site = "day", alpha = 1, observation = NULL)`). Otherwise the `unused argument` exception will be triggered.

Reply:
@clnsmth is the deprecation notice in roxygen? do you have an example?

Reply:
@sokole [Here's an example of argument deprecation in EMLassemblyline](https://github.com/EDIorg/EMLassemblyline/blob/master/R/make_eml.R#L228) where the `provenance` param is deprecated. The salient features are a warning message and passing of the deprecated argument to one of the supported params (in this case from `observation` to `data`).

Reply:
I'll handle this @sokole